ui/StatusQ/src/StatusQ/Popups/StatusMenu.qml

```diff
@@ -94,21 +84,34 @@ Menu {
         if (typeof closeHandler === "function") {
             closeHandler()
         }
+        d.maxDelegateImplWidth = 0
     }
 
     QtObject {
         id: d
         //helper property to get the max implicit width of the delegate
         property real maxDelegateImplWidth: 0
+
+        function calcMaxWidth() {
```

Comment:
This is not nice that we are forced here to compute the width manually. We make here assumptions about `menuItem` and it will probably fail when putting someting custom to the menu.

I was thinking about some alternatives. First try - using `menuItem.implicitWidth` - doesn't work because initial implicit width changes later causing the width too low. Binding to implicit width of every entry becomes complicated.

But the problem here lays in `contentItem` which is `ListView` as it's not able to compute implicit width based on implicit width of delegates.
Switching to container able to compute it's `implicitWidth` based on children's `implicitWidth` seems to solve the problem:

```qml
    contentItem: ColumnLayout {
        Repeater {
            model: root.contentModel

            onItemAdded: {
                item.Layout.fillWidth = true
                item.Layout.maximumWidth = root.maxImplicitWidth
            }
        }
    }
```
The problem is scrolling for very long menus (longer than window height). To support that, we can wrap into `StatusScrollView`:

```qml
    contentItem: StatusScrollView {
        padding: 0

        ColumnLayout {
            Repeater {
                model: root.contentModel

                onItemAdded: {
                    item.Layout.fillWidth = true
                    item.Layout.maximumWidth = root.maxImplicitWidth
                }
            }
        }
    }
```
It should render proper width out of the box, no manual computations.

Reply:
Additional benefit I observed is lack of blinking when opening first time in version using `calcMaxWidth` (probably caused by `Qt.callLater`).

Reply:
Nice idea! Much cleaner :)